src/js/port_handler.js

```diff
@@ -90,9 +90,6 @@ PortHandler.check_usb_devices = function (callback) {
             callback(self.dfu_available);
         }
         if (!$('option:selected', self.portPickerElement).data().isDFU) {
-            if (!GUI.connected_to) {
```

Comment:
Not sure why you want to remove this - to me it seems to be the right approach to reset the FC state whenever a disconnect / change of the selected port is detected.

Reply:
Because when configurator read hardwareId its not in Dfu mode, then it reset data and hardwareId never shown when is flashing. Im not sure how to fix this behavior.

Reply:
Maybe I can add a condition to aply only if not in flashing progress

Reply:
Ok, I can see what you are trying to do - the current implementation is correct (because there _is_ a disconnect when DFU mode is entered), but it is not super helpful, as the hardware id does not show during flashing.

> Maybe I can add a condition to aply only if not in flash progress

Yes, I think this makes sense - if flashing is in progress then we know that a 'disconnect' is not a physical disconnect.

Reply:
I got it!Definitely I have removed the reset whe chip erase is checked, as you said in the past it can be usefull if the user selects wrong target.